vispy/visuals/image.py

```diff
@@ -400,8 +400,14 @@ def _build_texture(self):
             # deal with clim on CPU b/c of texture depth limits :(
             # can eventually do this by simulating 32-bit float... maybe
             clim = self._clim
+            # deal with masked_array
+            if np.ma.is_masked(data):
+                mask = data.mask
+                data = np.array(data)
+                data[mask] = np.nan
+            is_num = np.isfinite(data)
             if isinstance(clim, string_types) and clim == 'auto':
-                clim = np.min(data), np.max(data)
+                clim = np.min(data[is_num]), np.max(data[is_num])
```

Comment:
`np.nanmin` / `np.nanmax`

Reply:
Yes but the array can contains `np.inf`. Actually, using `plt.imshow`, non-numerical pixels are not displayed.

Reply:
@EtienneCmb So you are saying in matplotlib `inf` values are ignored just like `nan`?

Reply:
Yes, `inf` and `nan` are ignored when plotting and the automatic `clim` is based on numerical values only.